docs/2.10.0/docs/concepts/ledger-model/upgrade-model-in-depth.rst

```diff
@@ -279,22 +279,55 @@ template on the left because it changes the type of ``x1`` from ``Int`` to ``Tex
 Template Keys
 ~~~~~~~~~~~~~
 
-The new version of a template cannot modify the key of the prior version of a
-template in any way.
+The key of the new version of a template must be a valid upgrade of the key of
+the prior version of that template.
 
-Adding a key leads to a validation error.
 
-Removing a key leads to a validation error.
-
-Changing the type of a key leads to a validation error.
-
-For Daml 2.9, key types can only use definitions from the current package or 
-from the Daml standard library. 
+Adding a key, removing a key, or changing the type of the key for a non-valid
+upgrade leads to a validation error.
```

Comment:
Can we express this without the use of "non-valid upgrade" (i.e. the negation)? I found that a quick read/parse of this statement quickly leads to misunderstandings.

Reply:
OR maybe we bring "non-valid upgrade" to the sentence start to make it more prominent?

Reply:
I broke it into two sentences.